bellows/zigbee/zcl/__init__.py

```diff
@@ -192,13 +192,22 @@ def write_attributes(self, attributes):
         for attrid, value in attributes.items():
             if isinstance(attrid, str):
                 attrid = self._attridx[attrid]
+            if attrid not in self.attributes:
+                self.error("%d is not a valid attribute id" % attrid)
```

Comment:
Formatting to logging should be with extra arguments, not by doing the formatting here -- that way a problem in doing the interpolation won't blow up the actual program, and the logging library will print out the error (without any extra handling on the client end).